### DISCUSSION TITLE
It's time to provide a status on new firmware release or progress!

### DISCUSSION THREAD

**Discussion by cpainchaud** [solokeys/solo2#117] [General]
Hi guys,

 Everyone here understands how many productions and logistic issues you had on the hardware side but now as thousands of us have received their key(s) we end up with a product that we can mostly not use because of many of the bugs with a few major FIDO2 enabled vendors.

 I am trying not to sounds like a whiner but bare minimum expected would be a page/thread/forum/idontknow where you communicate a bit on what is going on with the firmware. No communication is the worst way of doing anything. Let us know what is going on, good or bad... !

**Comment by piratenpanda** [comment]
The lack of communication is really frustrating

**Reply by piratenpanda** [reply]
q.e.d. unfortunately. As a backer this silence does not feel good. No new code for 2 months now, what's going on?

**Comment by needs-coffee** [comment]
I wonder if their dev has left the project. 
Looking across Solokeys' repos there doesn't seem to be any real code added since mid march (3 months now). Just some scattered documentation updates.

The fact that the software is this far behind in this project is both disappointing and alarming for the future of Solo. 
The hardware has been significantly delayed for reasons that have been previously documented in the Kickstarter updates and due to worldwide events that could not have been precedented, but despite the significant delay in hardware the firmware is still full of bugs and missing key features.  
There is next to no documentation for the product, your website 'quick start guide' and FAQ references mainly the first gen solo, and the cli tools for managing the key do not work reliably. 
In its current state this device is not fit to be relied on as a security device and it looks like I will be only be depending on my YubiKeys for the foreseeable future.

**Reply by cpainchaud** [reply]
there are still hope as NitroKeys is sharing the same framework and seem to be sliglty more active on the dev side

**Reply by PedroQ** [reply]
I wonder if it would be possible to flash a SoloKey with a NitroKey firmware and convert it.

**Comment by Pferdebockwurst** [comment]
I second this. The project looks pretty much dead for me.

**Reply by aussetg** [reply]
The project being dead is "fine" and unfortunately understandable given the situation... However we *really* need a last update that fix the key length and re-enable PIV.

After that they could relinquish control to someone else :/

To be faire there isn't any activity from non-Solo people either, so interest is pretty low...

**Reply by Pferdebockwurst** [reply]
I don't think it's fine. Also I hope the project only looks like it's dead, not that it is really dead.

And to be honest, I don't see why the situation should be understandable. What does the production and delivery issues have to do with the software development?

**Reply by KizzyCode** [reply]
What @Pferdebockwurst says, because *if* it's dead, we've crossed the line where this becomes fraud. You can still [order the Solokey 2 on Indiegogo](https://www.indiegogo.com/projects/solo-v2-safety-net-against-phishing/) with all the promises on the main page.

Of course it's always a possibility that Kickstarter campaigns fail or cannot deliver as promised – as a backer, thats calculated risk – but you must not continue to advertise goals once you know that you cannot/will not deliver them. So lets hope that this project is not dead... :/

**Reply by needs-coffee** [reply]
@aussetg which situation do you mean?
the point of a kickstarter is to raise funding for a product that will continue to be manufactured afterwards, so the firmware should be finished if they have any intent of that. I dont think you will find anyone jumping to buy another at the moment.
was the PIV issue as simple as key length? if so its strange they didnt just change that in an update rather than disable the feature.

**Reply by needs-coffee** [reply]
@KizzyCode i agree that Kick-starters are a calculated risk of backing a project not a guarantee of delivery - but continuing to sell them at this point on Indiegogo with that product description is, at a minimum, disingenuous, now bordering on fraudulent. 

I have been thinking about this "updatable firmware" selling point of the Solokeys and its a double edged sword. If you take a Yubikey - yes the firmware is fixed from the factory and can only be upgraded by replacing the device, but the device then has all the features at he point you purchase it, it it didn't, you would return it. With the solo and its upgradable firmware this has allowed them to ship an incomplete device and promise software updates later to add the missing functionality. 

The tricky bit if this project is "dead" is that the secure key's would not be upgradable would they? as the signing keys for the firmware are held by solo, so even if the firmware is upgraded by the community it would still need signed by solokeys in order to update them.

**Reply by aussetg** [reply]
@needs-coffee Chip shortage = can't make Solo2's = Can't sell Solo2 = Doesn't make money = Can't quit your job to work on the software fulltime ...

**Reply by piratenpanda** [reply]
> @needs-coffee Chip shortage = can't make Solo2's = Can't sell Solo2 = Doesn't make money = Can't quit your job to work on the software fulltime ...

If they'd communicate it like that, fine, but they don't which is not fine at all

**Comment by huaracheguarache** [comment]
Could @nickray or @conorpp please provide an update on the firmware update progress? I think it's unacceptable that we're being left in the dark like this.

**Comment by conorpp** [comment]
Hi everyone, sorry for infrequent communication -- lately we've gotten new machines at our manufacturing plant and I've been working on revising the designs for the new process.

The project is not abandoned.  I'm working on a firmware update now to cover the issues reported and will release it this coming week.

**Reply by Pferdebockwurst** [reply]
Now this is good news.

**Reply by thegcat** [reply]
Thank you for the work and for the update.

**Reply by synthgab** [reply]
If that's accurate, that's indeed good news...

However, based on expectations versus reality throughout all this project (with good reasons involved), please allow me to stay skeptical until I see a new published release which actually solves *all* the pending issues.

Thanks !

**Reply by synthgab** [reply]
It's already Friday of «this coming week» ... Just saying 👀

**Reply by cpainchaud** [reply]
@synthgab I am frustrated as much as you are but the way you are communicating may just make devs flee away for good. There is clearly some sort of distress causing this situation that we don't know about and I see no good reason for @conorpp to just drop an message saying he would make a release soon and go silent again for months.
While you might be right, you should be holding your negativity until the deadline has passed.

**Reply by synthgab** [reply]
In that case, they shouldn't mention any timeline and communicate more frequently... E.g. once a month on Kickstarter, as they did in the beginnings of the project, and which is reasonable.

Delays are understandable and accepted. Lack of communication and empty promises are not anymore. Github has been inactive for several months. I'm sorry, but right now, without any news from a shipment which should have been arrived a few weeks back nor any meanigful commit on GitHub, this project has to be considered dead until otherwise.

« We have a firmware upgrade that is nearing release...a few more bugs to iron out and then it will be made available for general use. »
April 2022: https://www.kickstarter.com/projects/conorpatrick/solo-v2-safety-net-against-phishing/posts/3484043

**Reply by john-prowse** [reply]
I can understand there is a fine balance between providing a product that people paid for (sometimes many) months ago 
vs 
improving firmware for people who have received hardware and want to use it.

I think most people can understand that covid and following supply chain crunch is unprecedented and time is money for everyone. 

If this project has a future for its maintainers, a realistic timeline for when firmware is going to be prioritised would go a long way to managing expectations for backers/future customers.

**Reply by kerkmann** [reply]
I can totally understand what @synthgab said. The main problem is really not the delays itself.
Things like that are happening, but the lack of communication is something, which isn't really great.
When I see a repository that is unmaintained for a long time and there is no mentioning or communication, it's really a red flag and let me think about alternatives...

No one is gonna shame you @conorpp if things do not go the way you think beforehand.
And I'm sure we all know sometimes there are bugs, which sounds trivial and can cause a lot of brain fart.

But letting people standing still without any communication is really problematic.
I've seen a lot of comments who says "solokeys is dead", "they are not developing it anymore" and all that rumors would be gone, if there would be a better communication.
I'm pretty sure all of you are doing a great job and all of you have a hard time.
But no one is seeing all your hard work, if the curtain is closed. ^^"

Thanks for all your hard work and I'm pretty the solo2 is rocking! :heart: 
Happy weekend! :partying_face:

**Reply by burtek** [reply]
From July 20th:

> ### Firmware
> This is another critical item for improvement.  We do have a fix for the Windows item... but there were a few more bugs to work out. Expecting to have a firmware release that will greatly improve our FIDO performance out in the very near future (1-3 weeks). This estimate has been pushed multiple times previously primarily due to the work crunch that has been experienced... we apologize for the delays involved.
> 
> We're targeting the FIDO Interoperability Event of end of September to receive the updated FIDO2 certification. If you're planning to be at the event let us know!

https://www.kickstarter.com/projects/conorpatrick/solo-v2-safety-net-against-phishing/posts/3561679/

So another 1-3 weeks.

**Comment by n00b42** [comment]
Just to have all related info here: In Kickstarter Update 10 days ago the following was stated:

> [...] Expecting to have a firmware release that will greatly improve our FIDO performance out in the very near future (1-3 weeks). [...]

**Reply by imilchev** [reply]
It's been 5 months without a commit. The key still has bugs in generating too long signatures which makes it useless with certain apps (like Keeper). I have to say that it looked like a very good initiative at the beginning but now it seems like a dead barely working prototype... I hope I am proven wrong but my patience is gone..

**Reply by kerkmann** [reply]
To be fair, it's still an early Kickstarter product. Like I said in my other posts, for me, it's totally fine that things are taking longer than expected. And I understand your frustration @imilchev , but I think it's also not fair to push them too much. :3

**Reply by burtek** [reply]
@kerkmann While I mostly agree with you, it's very bad approach to promise a deadline and not stick to it, without any proof that work has been done, which happened already multiple times. It would be MUCH better if we saw any changes in the repo and no sharp deadlines. I prefer "Work is in progress, we can't give an estimate yet, but here you can see latest commits" rather than "you'll get firmware in 1 week" which was posted 23 days ago, without any git changes visible

**Comment by Pferdebockwurst** [comment]
Guys, it's finally happening! The README.md has been updated!

**Reply by huaracheguarache** [reply]
It's now more than 3 weeks since the update on Kickstarter which said that the firmware update would be out in 1-3 weeks.

**Comment by huaracheguarache** [comment]
It's now 4 weeks since the announcement on Kickstarter that the firmware update would be out in the near future (1-3 weeks). Are you guys going to deliver anytime soon? This is beyond ridiculous.

**Comment by synthgab** [comment]
Woohoo !!
Some activity on the fido-authenticator repo ... Hopefully, things are actually moving forward 🤞

[Edit] typo

**Comment by mseri** [comment]
It seems to be out: https://github.com/solokeys/solo2/releases/tag/2.964.0 🥳 

Do you know how one can upgrade the firmware?

**Reply by Guldoman** [reply]
I used https://github.com/solokeys/solo2-cli
Then to actually update, after plugging in the key:
```bash
solo2 update
```

**Reply by TimWolla** [reply]
Just as a heads up. The update introduced a regression / data loss for me: https://github.com/solokeys/fido-authenticator/issues/19

**Reply by conorpp** [reply]
This shouldn't cause any regressions, but it will likely invalidate existing credentials you have on the key, be careful!

**Reply by thegcat** [reply]
Browsers on macOS won't recognise the key anymore:
- Safari: says "plug in your key and push the button", no effect, on GitHub it also locked up the browser on the window waiting for the key, the cancel button from that did not work.
- Firefox: Doesn't even try to look for a key, so I might have some config broken aside from the key.
- Chromium: also says "plug in your key and push the button", again nothing. Doesn't lock up though, so there's that.

**Reply by thegcat** [reply]
My iPhone will also only have the popup for NFC tags with URLs in it, that one pointing to solokeys.com, even when in a login flow in Safari.

**Reply by mseri** [reply]
On Mac it is working fine for me after the update, I just had to re-register the key

**Reply by burtek** [reply]
NFC works on android only if I remove cover from my phone, which is not ideal (although it works now, it didn't work at all previously)


Still can't login on android with key registered on windows. The other way around works.

Testing websites from https://github.com/solokeys/solo2/discussions/96#discussioncomment-2384191 now (mostly (lack of) support didn't change)

**Reply by bernhard-efler** [reply]
On linux the update does not work. When using `solo2 update` the key seems to hang and I get a timeout. Any ideas?

**Reply by huaracheguarache** [reply]
@bernhard-efler Try adding the following to your udev rules (it worked for me on Fedora 36): https://github.com/solokeys/solo2-cli/blob/main/70-solo2.rules

**Comment by nickray** [comment]
New release: https://github.com/solokeys/solo2/releases/tag/2.964.0
Associated discussion: https://github.com/solokeys/solo2/discussions/129
